Replace -f with -x for pytest tests.

pritamdamania87 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#46967 Replace -f with -x for pytest tests.**

Tests under `tests/distributed/_pipeline/sync` use pytest and
specifying the `-f` option for such tests as follows: `python test/run_test.py
-i distributed/_pipeline/sync/skip/test_api -- -f` doesn't work.

The equivalent option for pytest is `-x`. To resolve this issue, I've updated
`run_test.py` to replace `-f` with `-x` for pytest tests.

More details in https://github.com/pytorch/pytorch/issues/46782

#Closes: https://github.com/pytorch/pytorch/issues/46782

Differential Revision: [D24584556](https://our.internmc.facebook.com/intern/diff/D24584556/)

pritamdamania committed:
Replace -f with -x for pytest tests.

Tests under `tests/distributed/_pipeline/sync` use pytest and
specifying the `-f` option for such tests as follows: `python test/run_test.py
-i distributed/_pipeline/sync/skip/test_api -- -f` doesn't work.

The equivalent option for pytest is `-x`. To resolve this issue, I've updated
`run_test.py` to replace `-f` with `-x` for pytest tests.

More details in https://github.com/pytorch/pytorch/issues/46782

#Closes: https://github.com/pytorch/pytorch/issues/46782

Differential Revision: [D24584556](https://our.internmc.facebook.com/intern/diff/D24584556/)

[ghstack-poisoned]

malfet commented on test/run_test.py:
```diff
@@ -313,6 +313,13 @@ def run_test(test_module, test_directory, options, launcher_cmd=None, extra_unit
+
+    # If using pytest, replace -f with equivalent -x
+    if options.pytest:
+        for idx, arg in enumerate(unittest_args):
+            if arg == '-f':
+                unittest_args[idx] = '-x'
```
Wouldn't list comprehension make it a bit more concise here?
```suggestion
        unittest_args = [arg if arg != '-f' else '-x' for arg in unittest_args]
```

pritamdamania committed:
Update on "Replace -f with -x for pytest tests."


Tests under `tests/distributed/_pipeline/sync` use pytest and
specifying the `-f` option for such tests as follows: `python test/run_test.py
-i distributed/_pipeline/sync/skip/test_api -- -f` doesn't work.

The equivalent option for pytest is `-x`. To resolve this issue, I've updated
`run_test.py` to replace `-f` with `-x` for pytest tests.

More details in https://github.com/pytorch/pytorch/issues/46782

#Closes: https://github.com/pytorch/pytorch/issues/46782

Differential Revision: [D24584556](https://our.internmc.facebook.com/intern/diff/D24584556/)

[ghstack-poisoned]
